### PR TITLE
Fix unit testing

### PR DESCRIPTION
**Pull request by gabaldon**

### CODE REVIEW

File: src/store/uiInteractions.js [hunk 1]
```diff
@@ -1,7 +1,8 @@
-import { localStorageWrapper } from '@/main'
+import { LocalStorageWrapper } from '@/api'
 
 export default {
   state: {
+    localStorage: new LocalStorageWrapper(),
```
Comment (Tommytrg) [inline src/store/uiInteractions.js:5]: As the `localStorageWrapper` never changes it doesn't need to be inside the state

File: src/store/uiInteractions.js [hunk 2]
```diff
@@ -83,8 +84,8 @@ export default {
       // TODO: replace vue prototype https://vuejs.org/api/application.html#app-config-globalproperties
       // Vue.prototype.$notify(payload)
     },
-    saveShowModalAgain(_context, val) {
-      localStorageWrapper.setSkipSessionExpirationInfo(val)
+    saveShowModalAgain(context, val) {
+      context.state.localStorage.setSkipSessionExpirationInfo(val)
```
Comment (Tommytrg) [inline src/store/uiInteractions.js:88]:
```suggestion
  localStorageWrapper.setSkipSessionExpirationInfo(val)
```

File: src/main.ts [hunk 3]
```diff
@@ -1,20 +1,16 @@
...
-// import "~/styles/index.scss";
-// import 'uno.css
-import '~/styles/index.scss'
-import 'uno.css'
+import '@/styles/element-variables.scss'
+import '@/styles/index.scss'
 import '@/ipcHandlers'
 
 const api = new WalletApi()
-const localStorageWrapper = new LocalStorageWrapper()
```
Comment (Tommytrg) [inline src/main.ts:17]: I'd leave this as it is because doesn't need to be inside the store

File: test/unit/api.spec.js [hunk 4]
```diff
@@ -1,4 +1,5 @@
 import { standardizeTransactions } from '@/api'
+import { describe, expect, test } from 'vitest'
```
Comment (Tommytrg) [inline test/unit/api.spec.js:2]: if you don't want to write this line in every file we can set `globals` to `true`: https://vitest.dev/config/#globals

**Review by Tommytrg — CHANGES_REQUESTED** [review]

File: src/components/Transaction.vue [hunk 5]
```diff
@@ -44,7 +44,7 @@
         <div v-if="confirmed">
           <p data-test="time-ago" class="date">{{ timeAgo }}</p>
         </div>
-        <div v-else-if="!!Number(epoch)">
+        <div v-else-if="!!Number(epoch)" data-test="pending-confirmation">
```
[the remark this answers is not in the document]
Comment (gabaldon) [reply src/components/Transaction.vue:47]: This change is not related with the pr, should I updated in a refactor?

File: src/main.ts (hunk 3 above)
Comment (gabaldon) [reply src/main.ts:17]: This is not used in this file, we should not export it

File: src/components/OperatorOutput.vue [hunk 6]
```diff
@@ -12,22 +12,19 @@
...
         :content="stringifiedOutput"
       >
         <template #reference>
-          <p
-            v-if="explicitOutput"
-            data-test="output"
-            class="icon explicit-output"
-          >
+          <p v-if="explicitOutput" class="icon explicit-output">
+            >
```
[the remark this answers is not in the document]
Comment (gabaldon) [reply src/components/OperatorOutput.vue:23]: Because it can be text within the paragraph

File: src/main.ts [hunk 7]
```diff
@@ -1,28 +1,27 @@
...
 const eventProcessor = new ProcessWalletEvent()
+const localStorageWrapper = new LocalStorageWrapper()
 
 import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome'
 import { createApp } from 'vue'
 import App from './App.vue'
 
 const app = createApp(App)
 
+app.provide('localStorageWrapper', localStorageWrapper)
```
Comment (Tommytrg) [inline src/main.ts:24]:
```suggestion
```

File: src/main.ts [hunk 8]
```diff
@@ -1,28 +1,27 @@
...
-import '~/styles/index.scss'
-import 'uno.css'
+import '@/styles/element-variables.scss'
+import '@/styles/index.scss'
 import '@/ipcHandlers'
 
 const api = new WalletApi()
-const localStorageWrapper = new LocalStorageWrapper()
 const eventProcessor = new ProcessWalletEvent()
+const localStorageWrapper = new LocalStorageWrapper()
```
Comment (Tommytrg) [inline src/main.ts:16]:
```suggestion
```

File: tests/unit/src/background/utils/overwriteWitnetNodeConfiguration.spec.js [hunk 9]
```diff
@@ -1,17 +1,18 @@
...
 })
 
 describe('overwriteConfigFile', () => {
-  it('Should overwrite the configuration with a single url according to the parameters given', () => {
-    jest.spyOn(fs, 'writeFileSync').mockImplementation()
-    jest
-      .spyOn(fs, 'readFileSync')
-      .mockImplementation(() => 'node_url = "public_node_url3"')
+  test('Should overwrite the configuration with a single url according to the parameters given', () => {
+    vi.spyOn(fs, 'writeFileSync').mockImplementation
```
Comment (Tommytrg) [inline tests/unit/src/background/utils/overwriteWitnetNodeConfiguration.spec.js:12]:
```suggestion
```

File: tests/unit/src/components/CreateDataRequestForm.spec.js [hunk 10]
```diff
@@ -110,124 +106,73 @@ describe('CreateDataRequestForm.vue', () => {
...
-        .trigger('click')
-
+      const submitBtn = wrapper.find('[data-test="create-data-request-submit"]')
+      await submitBtn.trigger('click')
+      await flushPromises()
       expect(wrapper.emitted()['set-dr-values']).toBeTruthy()
     })
 
-    it('should show an error if the input cannot be converted to number', async () => {
+    test.skip('should show an error if the input cannot be converted to number', async () => {
```
Comment (Tommytrg) [inline tests/unit/src/components/CreateDataRequestForm.spec.js:125]: What's the problem with these test?

File: tests/unit/src/components/CreateDataRequestForm.spec.js [hunk 11]
```diff
@@ -110,124 +106,73 @@ describe('CreateDataRequestForm.vue', () => {
...
       wrapper.find('[data-test="collateral"]').setValue('10000000000')
-      await wrapper
-        .find('[data-test="create-data-request-submit"]')
-        .trigger('click')
+      const submitBtn = wrapper.find('[data-test="create-data-request-submit"]')
+      await submitBtn.trigger('click')
+      await flushPromises()
+      await flushPromises()
+      const error = wrapper.find('.el-form-item__error')
+      console.log(error)
```
Comment (Tommytrg) [inline tests/unit/src/components/CreateDataRequestForm.spec.js:140]:
```suggestion
```

File: tests/unit/src/components/ExportXprv.spec.js [hunk 12]
```diff
@@ -1,37 +1,52 @@
 import ExportXprv from '@/components/ExportXprv.vue'
+import Card from '@/components/card/Card.vue'
+import PasswordValidation from '@/components/PasswordValidation.vue'
 
-describe('ExportXprv', () => {
-  const exportMasterKeyMock = jest.fn()
-  const clearErrorMock = jest.fn()
-  const validatePasswordMock = jest.fn()
+import { describe, expect, test, vi } from 'vitest'
```
Comment (Tommytrg) [inline tests/unit/src/components/ExportXprv.spec.js:5]:
```suggestion
```

File: tests/unit/src/components/ResyncConfirmation.spec.js [hunk 13]
```diff
@@ -1,103 +1,125 @@
-import Dialog from 'element-ui/lib/dialog'
-import Button from 'element-ui/lib/button'
-
+import { flushPromises, mount } from '@vue/test-utils'
+import { describe, expect, test, vi } from 'vitest'
```
Comment (Tommytrg) [inline tests/unit/src/components/ResyncConfirmation.spec.js:2]:
```suggestion
```

File: tests/unit/src/components/SettingsResync.spec.js [hunk 14]
```diff
@@ -1,74 +1,81 @@
 import SettingsResync from '@/components/SettingsResync.vue'
 import { NETWORK_STATUS } from '@/constants'
 
+import { describe, expect, test, vi } from 'vitest'
+
```
Comment (Tommytrg) [inline tests/unit/src/components/SettingsResync.spec.js:5]:
```suggestion
```

File: tests/unit/src/components/TransactionList.spec.js [hunk 15]
```diff
@@ -1,7 +1,11 @@
 import TransactionList from '@/components/TransactionList.vue'
 import Transaction from '@/components/Transaction.vue'
-const setCurrentTransactionsPageActionMock = jest.fn()
-const setCurrentTransactionsPageMutationMock = jest.fn()
+
+import { describe, expect, test, vi } from 'vitest'
+
```
Comment (Tommytrg) [inline tests/unit/src/components/TransactionList.spec.js:5]:
```suggestion
```

File: tests/unit/src/components/TransactionList.spec.js [hunk 16]
```diff
@@ -109,64 +119,45 @@ describe('Changes the current page when clicking next button in pagination', ()
...
 
 describe('Renders the correct elements when click is not triggered', () => {
-  it('render the number of transactions', () => {
-    const wrapper = mount(TransactionList, getData())
+  test('render the number of transactions', () => {
+    const wrapper = mount(TransactionList, getData(87, 13))
 
     expect(wrapper.find('[data-test="transactions-length"]').text()).toContain(
-      'Transactions 87 transactions',
+      'Transactions87',
```
Comment (Tommytrg) [inline tests/unit/src/components/TransactionList.spec.js:126]: should this contain a white space?

Comment (gabaldon) [reply tests/unit/src/components/TransactionList.spec.js:126]: No, becase it is a span that has a padding or a gap, it is not necessary

**Review by Tommytrg — APPROVED** [review]
:fire:

**Pull request merged**